maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/SurefireDependencyResolver.java

```diff
@@ -61,20 +62,21 @@
 {
     static final String PROVIDER_GROUP_ID = "org.apache.maven.surefire";
 
-    private static final String[] PROVIDER_CLASSPATH_ORDER = {
-        "surefire-junit3",
-        "surefire-junit4",
-        "surefire-junit47",
-        "surefire-testng",
-        "surefire-junit-platform",
-        "surefire-api",
-        "surefire-logger-api",
-        "surefire-shared-utils",
-        "common-java5",
-        "common-junit3",
-        "common-junit4",
-        "common-junit48",
-        "common-testng-utils"
+    private static final String[][] PROVIDER_CLASSPATH_ORDER = {
+        // groupId, artifactId [, classifier]
+        {"org.apache.maven.surefire", "surefire-junit3"},
+        {"org.apache.maven.surefire", "surefire-junit4"},
+        {"org.apache.maven.surefire", "surefire-junit47"},
+        {"org.apache.maven.surefire", "surefire-testng"},
+        {"org.apache.maven.surefire", "surefire-junit-platform"},
+        {"org.apache.maven.surefire", "surefire-api"},
+        {"org.apache.maven.surefire", "surefire-logger-api"},
+        {"org.apache.maven.surefire", "surefire-shared-utils"},
+        {"org.apache.maven.surefire", "common-java5"},
+        {"org.apache.maven.surefire", "common-junit3"},
+        {"org.apache.maven.surefire", "common-junit4"},
+        {"org.apache.maven.surefire", "common-junit48"},
+        {"org.apache.maven.surefire", "common-testng-utils"}
```

Comment:
Is it related to subject of PR?

Reply:
yes, I know, this was related to the thinking process with classifier.
Let me pls squash the commits, it's easier to avoid changing this class afterwards.